lib/include/jxl/decode.h

```diff
@@ -913,6 +953,144 @@ JXL_EXPORT JxlDecoderStatus JxlDecoderSetJPEGBuffer(JxlDecoder* dec,
+ * JxlDecoderReleaseBoxBuffer is called or the decoder is destroyed or reset so
+ * must be kept alive until then.
+ *
+ * @param dec decoder object
+ * @param data pointer to next bytes to write to
+ * @param size amount of bytes available starting from data
+ * @return JXL_DEC_ERROR if output buffer was already set and
+ * JxlDecoderReleaseBoxBuffer was not called on it, JXL_DEC_SUCCESS otherwise
+ */
+JXL_EXPORT JxlDecoderStatus JxlDecoderSetBoxBuffer(JxlDecoder* dec,
+                                                   uint8_t* data, size_t size);
+
+/**
+ * Releases buffer which was provided with JxlDecoderSetBoxBuffer.
+ *
+ * Calling JxlDecoderReleaseBoxBuffer is required whenever
+ * a buffer is already set and a new buffer needs to be added with
+ * JxlDecoderSetBoxBuffer, but is not required before JxlDecoderDestroy or
+ * JxlDecoderReset.
+ *
+ * Calling JxlDecoderReleaseBoxBuffer when no buffer is set is
+ * not an error and returns 0.
+ *
+ * @param dec decoder object
+ * @return the amount of bytes the decoder has not yet written to of the data
+ * set by JxlDecoderSetBoxBuffer, or 0 if no buffer is set or
+ * JxlDecoderReleaseBoxBuffer was already called.
+ */
+JXL_EXPORT size_t JxlDecoderReleaseBoxBuffer(JxlDecoder* dec);
+
+/**
+ * Configures whether to get boxes in raw mode or in uncompressed
+ * mode.
+ *
+ * The default mode is non-raw (JXL_FALSE), where "brob" boxes are transparently
+ * decompressed. This setting can only be changed
+ * before decoding, or directly after a JXL_DEC_BOX event, and is preserved
+ * otherwise.
+ *
+ * @param dec decoder object
+ * @param enable_raw whether to enable or disable raw mode
+ * @return JXL_DEC_ERROR if input was already set without releasing,
+ * JXL_DEC_SUCCESS otherwise
+ */
+JXL_EXPORT void JxlDecoderSetBoxModeRaw(JxlDecoder* dec, JXL_BOOL enable_raw);
+
+/**
+ * Outputs the type of the current box, after a JXL_DEC_BOX event occured, as 4
+ * characters plus a null termination character. In case of a compressed box,
+ * this returns the type of the underlying uncompressed box, not "brob". Use
+ * @see JxlDecoderGetBoxTypeRaw to get the raw type.
+ *
+ * @param dec decoder object
+ * @param type buffer to copy the type into
+ * @return JXL_DEC_SUCCESS if the value is available, JXL_DEC_ERROR if not, for
+ *    example the JXL file does not use the container format.
+ */
+JXL_EXPORT JxlDecoderStatus JxlDecoderGetBoxType(JxlDecoder* dec, char type[5]);
+
+/**
+ * Outputs whether the given type equals the type returned by
+ * JxlDecoderGetBoxType.
+ *
+ * @param dec decoder object
+ * @param type box type to compare with, e.g. "Exif". Typenames are 4 non-null
+ * characters, passing a const char* string with null terminator is possible but
+ * not required.
+ * @return JXL_TRUE if the box type is available and equal to the given type,
+ * JXL_FALSE otherwise.
+ */
+JXL_EXPORT JXL_BOOL JxlDecoderBoxTypeEquals(JxlDecoder* dec,
```

Comment:
Why do we need this function if we have a getter for the box type? isn't this just memcmp() of the GgetBoxType value?

Reply:
This is for convenience because in 99% of the usecases, you only want to compare to a type, like `JxlDecoderBoxTypeEquals("Exif", dec)`, which saves a temporary variable and C style string comparison

However, both you and Jon pointed out the redundancy, so I can remove this function after all and make the API more barebones